drivers/usb/device/usb_dc_stm32.c

```diff
@@ -321,7 +323,7 @@ static uint32_t usb_dc_stm32_get_maximum_speed(void)
 	if (!strncmp(USB_MAXIMUM_SPEED, "high-speed", 10)) {
 		speed = USB_OTG_SPEED_HIGH;
 	} else if (!strncmp(USB_MAXIMUM_SPEED, "full-speed", 10)) {
-#if USB_OTG_HS_EMB_PHY
+#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(USB_OTG_HS_EMB_PHY)
 		speed = USB_OTG_SPEED_HIGH_IN_FULL;
 #else
```

Comment:
This part is a little crap (using series Kconfig flag to modify dts extracted value), but I don't see better way to do it without modifying the code around.

Reply:
Also, this, and some other parts, would need to get refactored to support actually running the device in high speed mode.